packages/http-client-java/generator/http-client-generator-clientcore-test/Generate.ps1

```diff
@@ -0,0 +1,122 @@
+  # Without this, we could have multiple runs trying to write to the same directory which introduces race conditions.
+  $tspOptions = "--option ""@typespec/http-client-java.emitter-output-dir={project-root}/tsp-output/$(Get-Random)"""
+  if ($tspFile -match "type[\\/]enum[\\/]extensible[\\/]") {
+    # override namespace for reserved keyword "enum"
+    $tspOptions += " --option ""@typespec/http-client-java.namespace=type.enums.extensible"""
+  } elseif ($tspFile -match "type[\\/]enum[\\/]fixed[\\/]") {
+    # override namespace for reserved keyword "enum"
+    $tspOptions += " --option ""@typespec/http-client-java.namespace=type.enums.fixed"""
+  } elseif ($tspFile -match "type[\\/]array" -or $tspFile -match "type[\\/]dictionary") {
+    # TODO https://github.com/Azure/autorest.java/issues/2964
+    # also serve as a test for "use-object-for-unknown" emitter option
+    $tspOptions += " --option ""@typespec/http-client-java.use-object-for-unknown=true"""
+  }
+
+  $tspTrace = "--trace import-resolution --trace projection --trace http-client-java"
+  $tspCommand = "npx --no-install tsp compile $tspFile $tspOptions $tspTrace"
+
+  $timer = [Diagnostics.Stopwatch]::StartNew()
+  $generateOutput = Invoke-Expression $tspCommand
+  $timer.Stop()
+
+  $global:ExitCode = $global:ExitCode -bor $LASTEXITCODE
+
+  if ($LASTEXITCODE -ne 0) {
+    Write-Host "
+  ========================
+  $tspCommand
+  ========================
+  FAILED (Time elapsed: $($timer.ToString()))
+  $([String]::Join("`n", $generateOutput))
+    "
+  } else {
+    Write-Host "
+  ========================
+  $tspCommand
+  ========================
+  SUCCEEDED (Time elapsed: $($timer.ToString()))
+    "
+  }
+
+  if ($global:ExitCode -ne 0) {
+    exit $global:ExitCode
+  }
+}
+
+./Setup.ps1
+
+if (Test-Path ./src/main) {
+  Remove-Item ./src/main -Recurse -Force
+}
+if (Test-Path ./src/samples) {
+  Remove-Item ./src/samples -Recurse -Force
+}
+if (Test-Path ./tsp-output) {
+  Remove-Item ./tsp-output -Recurse -Force
+}
+
+# generate for http-specs/azure-http-specs test sources
+Copy-Item -Path node_modules/@typespec/http-specs/specs -Destination ./ -Recurse -Force
+# remove xml tests, emitter has not supported xml model
+Remove-Item ./specs/payload/xml -Recurse -Force
+
+# ServiceVersion not in core
+Remove-Item ./specs/versioning -Recurse -Force
+Remove-Item ./specs/server/path -Recurse -Force
+Remove-Item ./specs/server/versions -Recurse -Force
+# TokenCredential not in core
+Remove-Item ./specs/authentication/oauth2 -Recurse -Force
+Remove-Item ./specs/authentication/union -Recurse -Force
+# Base64Url not in core
+Remove-Item ./specs/encode/bytes -Recurse -Force
+# DateTimeRfc1123 is private in beta.1, should now be public in main
+Remove-Item ./specs/encode/datetime -Recurse -Force
+Remove-Item ./specs/special-headers -Recurse -Force
+# JacksonAdapter not in core
+Remove-Item ./specs/encode/duration -Recurse -Force
```

Comment:
At present, lots of tests is skipped, due to no class in clientcore.

Some tests are also disabled.

We will later make a summary on these, and discuss what need to be fixed in clientcore, what need to be handled different in emitter.